### PR TITLE
Tweak tests on top of pull #608

### PR DESCRIPTION
**Pull request by daviddrysdale**
Pull #608 is nearly done, but a last-minute tweak caused a build problem.  To save another round-trip, I've taken #608 without the last commit, then added a couple of commits to sort out the tests.

So only the last 2 commits really need reviewing.

### CODE REVIEW

File: examples/ct/cert_checker_test.go [hunk 1]
```diff
@@ -23,154 +23,133 @@ import (
...
-		t.Fatal("failed to load fake root")
-	}
-
-	_, err := ValidateChain(jsonChain, *trustedRoots, false)
-
-	if err == nil {
-		t.Fatal("verification accepted an invalid chain (ordering)")
-	}
+func makePoisonNonCritical(cert *x509.Certificate) *x509.Certificate {
+	// Poison extension needs to be marked as criticial to qualify as a pre-cert.
```
Comment (Martin2112) [inline examples/ct/cert_checker_test.go]: Just for readability can you comment that it's creating an invalid pre-cert as in the method below. When I read "needs to be marked as be critical" and saw Critical:false  I had to think about it.

File: examples/ct/instance.go [hunk 2]
```diff
@@ -82,6 +84,21 @@ func LogConfigFromFile(filename string) ([]LogConfig, error) {
 	return cfg, nil
 }
 
+var stringToKeyUsage = map[string]x509.ExtKeyUsage{
+	"Any":                        x509.ExtKeyUsageAny,
```
Comment (Martin2112) [inline examples/ct/instance.go]: You'd think this belonged with the X.509 code. I assume they don't have it.

File: examples/ct/cert_checker_test.go (hunk 1 above)
Comment (daviddrysdale) [inline examples/ct/cert_checker_test.go]: Done.

File: examples/ct/instance.go (hunk 2 above)
Comment (daviddrysdale) [inline examples/ct/instance.go]: Correct.

**Pull request merged**